tests/testing_common.py

```diff
@@ -304,6 +305,44 @@ def make_inputs_require_grad(module, input, output):
 
         assert dummy_output.requires_grad
 
+    def _test_load_model_empty_weights(self, model_id, config_cls, config_kwargs):
+        # Ensure that init_empty=True works for from_pretrained and load_adapter and that the resulting model's
+        # parameters are on the correct device.
+        model = self.transformers_class.from_pretrained(model_id).to(self.torch_device)
+        config = config_cls(
+            base_model_name_or_path=model_id,
+            **config_kwargs,
+        )
+        model = get_peft_model(model, config)
+
+        # note: not using the context manager here because it fails on Windows CI for some reason
+        tmp_dirname = tempfile.mkdtemp()
+        try:
+            model.save_pretrained(tmp_dirname)
+
+            model = self.transformers_class.from_pretrained(model_id).to(self.torch_device)
+            model = PeftModel.from_pretrained(model, tmp_dirname, torch_device=self.torch_device, init_empty=True)
+            assert {p.device.type for p in model.parameters()} == {self.torch_device}
+
+            model.load_adapter(tmp_dirname, adapter_name="other", init_empty=True)
+            assert {p.device.type for p in model.parameters()} == {self.torch_device}
+        finally:
+            try:
+                shutil.rmtree(tmp_dirname)
+            except PermissionError:
+                # windows error
+                pass
+
+        # also test injecting directly
+        del model
+        model = self.transformers_class.from_pretrained(model_id).to(self.torch_device)
```

Comment:
Would it make sense to do this for a diffusers model class as well?

Reply:
Not sure if we would expose another argument for `low_cpu_mem_usage` for loading LoRAs. Would prefer to keep the `peft` level atomic feature tests within `peft`, itself. 

If the test suite needs a big rewrite, we could just create a Tester class with a specific model class from `diffusers` and write a simple test case like this one. Will that work?

Reply:
I added dedicated stable diffusion tests. These tests should more or less reflect how the code in diffusers would look like so as to use this new feature.

https://github.com/huggingface/peft/pull/1961/commits/fbace51d7f8dcd111da334e429aeb87fe8c9b70e